src/votekit/pref_profile/pref_profile.py

```diff
@@ -891,6 +885,9 @@ def __init_formatted_df(
         if self.candidates != tuple() and contains_scores_indicator:
             col_order = list(self.candidates) + temp_col_order
         elif contains_scores_indicator:
+            for c in candidates_cast:
+                if c not in df.columns:
+                    df[c] = [np.nan] * len(df)
```

Comment:
Something like this. I don't remember if that syntax is valid. This just makes it clear what we are doing for the remaining candidates

```suggestion
                remaining_cands = set(candidates_cast) - set(df.columns)
                empty_df_cols = np.array(len(df), len(remaining_cands), fill=np.nan)
                df[list(remaining_cands)] = empty_df_cols
```